remote-api/src/main/java/org/alfresco/rest/api/impl/TagsImpl.java

```diff
@@ -104,6 +104,8 @@ public List<Tag> addTags(String nodeId, final List<Tag> tags)
 			{
 				throw new UnsupportedResourceOperationException("Cannot tag this node");
 			}
+			List<Pair<String, Integer>> tagsByCount = null;
+			Map<String, Integer> tagsByCountMap = new HashMap<String, Integer>();
```

Comment:
Neither of these things are named clearly. They both look likely to contain the same information, so maybe we don't need both. I think a better name might be `tagsAndCounts` or `tagCountMap`.

Reply:
Yes - looking below I can see that the list shouldn't be defined here.

Reply:
done